header/header.js

```diff
@@ -0,0 +1,141 @@
+import {LitElement, html} from 'https://unpkg.com/@polymer/lit-element@latest/lit-element.js?module';
+
+const debug = false;
+
+class BULHeader extends LitElement {
+
+  constructor(){
+    super();
+  }
+
+  // don't need 'slot' functionality, so lets use Light DOM
+  createRenderRoot(){ return this; }
+
+  /** store information on the current page */
+  static get properties() {
+    return {
+      curr_url: {type: String, notify:true},
+      curr_primary: {type: String}, // research, services, about, help
+      curr_secondary: {type: String}, // guides, help, [library-names]
+      curr_search: {type: String}, // primo, guides, wp, faq, ...
+      str_options: {type: String}, // cascade str_options down to search
+      logged_in: {type: Boolean}
+    };
+  }
+
+  /** render the html (with 'bulib-search' wc) to the page  */
+  render() {
+    this._setCurrSiteInfo();
+    let secondaryNavMain = (this.curr_library)? html`<bulib-libsel library="${this.curr_library}"></bulib-libsel>` : html`<h1>Subsite: ${this.curr_secondary}</h1>`;
```

Comment:
filling this in for each site should be out of scope for this element, and taken care of via a `<slot>`. 

doing so requires using the shadow DOM, which broke the reactive elements. will need to be revisited later.